File: scraper/job.py
class Job:
    id = ""
    title = ""
    description = ""
    requirements = ""
    contact : str | None = ""
    phone = ""
    email = ""
    company : str | None = ""
    location = ""
    postedOn = ""
    postedOnRaw = 0
    hours = ""
    wage = ""
    start = ""
    duration = ""
    apply = ""
    deadline = ""
    comments = ""

    def __init__(self, url):
        self.url = url

    def to_dict(self):
        return {
            'id' : self.id,
            'url' : self.url,
            'title' : self.title,
            'description' : self.description,
            'requirements' : self.requirements,
            'contact' : self.contact,
            'phone' : self.phone,
            'email' : self.email,
            'company' : self.company,
            'location' : self.location,
            'postedOn' : self.postedOn,
            'postedOnRaw' : self.postedOnRaw,
            'hours' : self.hours,
            'wage' : self.wage,
            'start' : self.start,
            'duration' : self.duration,
            'apply' : self.apply,
            'deadline' : self.deadline,
            'comments' : self.comments
        } 
    
    def from_dict(self, d: dict):
        self.id = str(d.get('id', ""))
        self.title = str(d.get('title', ""))
        self.description = str(d.get('description', ""))
        self.requirements = str(d.get('requirements', ""))
        self.contact = str(d.get('contact', ""))
        self.phone = str(d.get('phone', ""))
        self.email = str(d.get('email', ""))
        self.company = str(d.get('company', ""))
        self.location = str(d.get('location', ""))
        self.postedOn = str(d.get('postedOn', ""))
        self.postedOnRaw = int(d.get('postedOnRaw', 0))
        self.hours = str(d.get('hours', ""))
        self.wage = str(d.get('wage', ""))
        self.start = str(d.get('start', ""))
        self.duration = str(d.get('duration', ""))
        self.apply = str(d.get('apply', ""))
        self.deadline = str(d.get('deadline', ""))
        self.comments = str(d.get('comments', ""))
        # Ensure the URL is always a string
        self.url = str(d.get('url', self.url))

File: JobSite/Models/Job.cs
namespace JobSite.Models
{
    public class Job
    {
        public string? Id { get; set; } = "";
        public string? Url { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public string? Requirements { get; set; }
        public string? Contact { get; set; }
        public string? Phone { get; set; }
        public string? Email { get; set; }
        public string? Company { get; set; }
        public string? Location { get; set; }
        public string? PostedOn { get; set; }
        public string? Hours { get; set; }
        public string? Wage { get; set; }
        public string? Start { get; set; }
        public string? Duration { get; set; }
        public string? Apply { get; set; }
        public string? Deadline { get; set; }
        public string? Comments { get; set; }

        public void FromDict(Dictionary<string, object> dict)
        {
            Id = dict.TryGetValue("id", out object? id) && id != null ? id.ToString() : string.Empty;
            Title = dict.TryGetValue("title", out object? title) && title != null ? title.ToString() : null;
            Description = dict.TryGetValue("description", out object? description) && description != null ? description.ToString() : null;
            Requirements = dict.TryGetValue("requirements", out object? requirements) && requirements != null ? requirements.ToString() : null;
            Contact = dict.TryGetValue("contact", out object? contact) && contact != null ? contact.ToString() : null;
            Phone = dict.TryGetValue("phone", out object? phone) && phone != null ? phone.ToString() : null;
            Email = dict.TryGetValue("email", out object? email) && email != null ? email.ToString() : null;
            Company = dict.TryGetValue("company", out object? company) && company != null ? company.ToString() : null;
            Location = dict.TryGetValue("location", out object? location) && location != null ? location.ToString() : null;
            PostedOn = dict.TryGetValue("posted_on", out object? posted_on) && posted_on != null ? posted_on.ToString() : null;
            Hours = dict.TryGetValue("hours", out object? hours) && hours != null ? hours.ToString() : null;
            Wage = dict.TryGetValue("wage", out object? wage) && wage != null ? wage.ToString() : null;
            Start = dict.TryGetValue("start", out object? start) && start != null ? start.ToString() : null;
            Duration = dict.TryGetValue("duration", out object? duration) && duration != null ? duration.ToString() : null;
            Apply = dict.TryGetValue("apply", out object? apply) && apply != null ? apply.ToString() : null;
            Deadline = dict.TryGetValue("deadline", out object? deadline) && deadline != null ? deadline.ToString() : null;
            Comments = dict.TryGetValue("comments", out object? comments) && comments != null ? comments.ToString() : null;
        }
    }
}
